source/configuration/templates.rst

```diff
@@ -631,31 +631,82 @@ Do NOT use them if, otherwise you may receive a conflict in the future
+used with options: ``$SpaceLFOnReceive on``,
+``$EscapeControlCharactersOnReceive off``, ``$DropTrailingLFOnReception off``,
+the log format will conform to sysklogd log format.
+
+   "%TIMESTAMP% %HOSTNAME% %syslogtag%%msg:::sp-if-no-1st-sp%%msg%\n"
+
+**RSYSLOG_ForwardFormat** - a new high-precision forwarding format very
+similar to the traditional one, but with high-precision timestamps
+and timezone information. Recommended to be used when sending
+messages to rsyslog 3.12.5 or above.
+
+   "<%PRI%>%TIMESTAMP:::date-rfc3339% %HOSTNAME% %syslogtag:1:32% %msg:::sp-if-no-1st-sp%%msg%"
+
+**RSYSLOG_SyslogProtocol23Format** - the format specified in IETF's
+internet-draft ietf-syslog-protocol-23, which is very close to the actual
+syslog standard `RFC5424 <https://tools.ietf.org/html/rfc5424>`_ (we couldn't
+update this template as things were in production for quite some time when
+RFC5424 was finally approved). This format includes several improvements.
+You may use this format with all relatively recent versions of rsyslog or syslogd.
+
+   "<%PRI%>1 %TIMESTAMP:::date-rfc3339% %HOSTNAME% %APP-NAME% %PROCID% %MSGID% %STRUCTURED-DATA% %msg%\n"
+
+**RSYSLOG_DebugFormat** - a special format used for troubleshooting
+property problems. This format is meant to be written to a log file.
+Do **not** use for production or remote forwarding.
+
+   "Debug line with all properties:\nFROMHOST: '%FROMHOST%', fromhost-ip: '%fromhost-ip%', HOSTNAME: '%HOSTNAME%', PRI: %PRI%,\nsyslogtag '%syslogtag%', programname: '%programname%', APP-NAME: '%APP-NAME%', PROCID: '%PROCID%', MSGID: '%MSGID%',\nTIMESTAMP: '%TIMESTAMP%', STRUCTURED-DATA: '%STRUCTURED-DATA%',\nmsg: '%msg%'\nescaped msg: '%msg:::drop-cc%'\ninputname: %inputname% rawmsg: '%rawmsg%'\n$!:%$!%\n$.:%$.%\n$/:%$/%\n\n"
+
+**RSYSLOG_WallFmt** - Contains information about the host and the time the
+message was generated and at the end the syslogtag and message itself.
+
+   "\r\n\7Message from syslogd@%HOSTNAME% at %timegenerated% ...\r\n%syslogtag%%msg%\n\r"
+
+**RSYSLOG_StdUsrMsgFmt** - The syslogtag followed by the message is returned.
+
+   " %syslogtag%%msg%\n\r"
+
+**RSYSLOG_StdDBFmt** - Generates a insert command with the message
+properties, into table SystemEvents for a mysql database.
+
+   "\"insert into SystemEvents (Message, Facility, FromHost, Priority, DeviceReportedTime, ReceivedAt, InfoUnitID, SysLogTag) values ('%msg%', %syslogfacility%, '%HOSTNAME%', %syslogpriority%, '%timereported:::date-mysql%', '%timegenerated:::date-mysql%', %iut%, '%syslogtag%')\",SQL"
+
+**RSYSLOG_StdPgSQLFmt** - Generates a insert command with the message
+properties, into table SystemEvents for a pgsql database.
+
+   "\"insert into SystemEvents (Message, Facility, FromHost, Priority, DeviceReportedTime, ReceivedAt, InfoUnitID, SysLogTag) values ('%msg%', %syslogfacility%, '%HOSTNAME%', %syslogpriority%, '%timereported:::date-pgsql%', '%timegenerated:::date-pgsql%', %iut%, '%syslogtag%')\",STDSQL"
```

Comment:
This is the C string from the source. To obtain the actual template definition, it needs to be unescaped.